apm-agent-plugins/apm-asynchttpclient-plugin/src/test/java/co/elastic/apm/agent/asynchttpclient/AsyncHttpClientInstrumentationTest.java

```diff
@@ -49,11 +46,32 @@ public AsyncHttpClientInstrumentationTest(RequestExecutor requestExecutor) {
         this.requestExecutor = requestExecutor;
     }
 
+    public static AsyncHandler<Response> customAsyncHandler = new AsyncCompletionHandler<Response>() {
+        @Override
+        public State onStatusReceived(HttpResponseStatus responseStatus) {
+            assert(tracer.currentTransaction() != null);
```

Comment:
Not sure how to add a proper test making sure the context in the methods are active or not.